packages/block-library/src/image/index.php

```diff
@@ -28,12 +28,33 @@ function render_block_core_image( $attributes, $content, $block ) {
 		return '';
 	}
 
-	if ( isset( $attributes['data-id'] ) ) {
```

Comment:
The one thing I'm nervous about in this PR is removing the use of the temporary `data-id` block attribute, as plugins could be filtering it to a different value. Though I don't know why they would do that.

Unfortunately it's very difficult to check whether that might be the case.

I do think it's a bit of tech debt—particularly the way it causes the block's `id` is being stored in two attributes and the camel casing of the attribute name. I can add a dev note, but keen to get the thoughts of others.

Reply:
It is a breaking change 😢 . But filtering a temporary value is also not a really good development choice, right?

Reply:
As explained in [this comment](https://github.com/WordPress/gutenberg/pull/63013#issuecomment-2202771482), I made a commit with a potential alternative approach that doesn't require removing the temporary `data-id` block attribute. It seems to work and it seems closer to the existing implementation.

Reply:
Thanks for pushing a commit. I did consider something similar to that, and I think it's not a bad choice, but still has some drawbacks.

With the new changes, the image block's `id`  attribute (`$rendered_id`) is still being used for the `data-id` html attribute, whereas in trunk it's the injected `data-id`, so I don't think it completely addresses the back compat concerns. One use case for filtering `data-id` would be to modify the html attribute to something else, and that would no longer work. My conclusion is that we can't really support `data-id` (the block attribute) properly, and that's why I lean towards removing it in favor of the regular `id` (if it's going to break, then it's a good opportunity to clean up the code as much as possible).

The changes also seem to couple the `wp-image-` classname to the `data-id` attribute (the whole logic is in the `isset( $attributes['data-id'] )` if statement), which wasn't the case before. `data-id` is only present for image blocks in galleries, so this would result in the classname only being replaced for gallery images and not standalone images. A small change would be needed to fix that.